spec.html

```diff
@@ -113,16 +110,25 @@ <h1>`Promise.any` Reject Element Functions</h1>
       1. Set _errors_[_index_] to _x_.
       1. Set _remainingElementsCount_.[[Value]] to _remainingElementsCount_.[[Value]] - 1.
       1. If _remainingElementsCount_.[[Value]] is 0, then
-        1. Let _errorsArray_ be CreateArrayFromList(_errors_).
-        1. Let _error_ be a newly created `AggregateError` object.
-        1. Perform ! CreateDataProperty ( _error_, `"errors"`, _errorsArray_ ).
-        1. Return ThrowCompletion ( _error_ ).
+        1. Return ThrowAggregateError(_errors_).
       1. Return *undefined*.
     </emu-alg>
     <p>The `"length"` property of a `Promise.any` resolve element function is 1.</p>
   </emu-clause>
 </emu-clause>
 
+<emu-clause id="sec-promise.any-throwaggregateerror">
+  <h1>ThrowAggregateError ( _errors_ )</h1>
+  <p>The abstract operation ThrowAggregateError(_errors_) creates and throws an instance of AggregateError</p>
+  <emu-alg>
+    1. Let _errorsArray_ be CreateArrayFromList(_errors_).
+    1. Let _error_ be a newly created `AggregateError` object.
+    1. Let _errorsDesc_ be the PropertyDescriptor { [[Value]]: _errorsArray_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.
```

Comment:
Is there a reason not to just use `? Construct(%AggregateError%, « errors »)`, since the %AggregateError% constructor already has the defined semantics?

Reply:
what about the `message` argument?

Reply:
We don't seem to care about `message` elsewhere in the spec as the actual message is implementation dependent and may even be localised, yet this isn't formalized anywhere in the spec for the numerous "throw a **TypeError** exception" cases. I was wondering if we need to be more formal about this in general. Something like:

1. Let _message_ be a host implementation defined message.
1. Perform ? Throw(%Error%, &laquo; _message_ &raquo;).

Reply:
i suspect creation of errors typically avoids using Construct precisely to avoid the need for that formalism.

Reply:
Not just `message`, but also `stack`, `line`, and any other non-standard properties.